docs/control-structures.rst

```diff
@@ -29,6 +29,11 @@ something like::
         }
     }
 
+.. note::
```

Comment:
I still think this is the wrong place. Such types are perfectly fine for internal functions and such a restriction that will be lifted in the future might confuse people if mentioned here.

Reply:
@chriseth OK, but after a quick search I don't think it's mentioned anywhere else, so I'll remove and stash for a future PR after finding a suitable place for the text.